### PR TITLE
app-shells/starship: Add 1.16.0

### PR DESCRIPTION
**Pull request by xz-dev**
Closes: [Bug 911510](https://bugs.gentoo.org/911510)

### CODE REVIEW

**Comment by xz-dev** [comment]
It's there anything need fix?

**Review by bowlofeggs — APPROVED** [review]
LGTM. I don't have commit privileges, so we will have to wait for someone to merge.

File: app-shells/starship/starship-1.16.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,432 @@
+# Copyright 2022-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+CRATES="
```
Comment (arthurzam) [inline app-shells/starship/starship-1.16.0.ebuild:6]: Please use newer syntax of CRATES, `name@version`, which results in:

```
CRATES="
    adler@1.0.2
    ...
"
```

For future bumps, you can use pycargoebuild which is the new best tool for rust ebuilds.

File: app-shells/starship/starship-1.16.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,432 @@
...
+HOMEPAGE="https://starship.rs/"
+SRC_URI="
+	https://github.com/starship/starship/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz
+	${CARGO_CRATE_URIS}
+"
+
+LICENSE="0BSD Apache-2.0 Apache-2.0-with-LLVM-exceptions BSD BSD-2 Boost-1.0 CC0-1.0 ISC MIT MIT-0 MPL-2.0 Unicode-DFS-2016 Unlicense WTFPL-2 ZLIB"
+
+SLOT="0"
+KEYWORDS="~amd64"
```
Comment (arthurzam) [inline app-shells/starship/starship-1.16.0.ebuild:401]: Was there a reason to drop `~arm64`?

**Review by arthurzam — CHANGES_REQUESTED** [review]
Please rebase over master, there are conflicts now sadly.

File: app-shells/starship/starship-1.16.0.ebuild (hunk 2 above)
Comment (xz-dev) [reply app-shells/starship/starship-1.16.0.ebuild:401]: I think drop it is fine.

File: app-shells/starship/starship-1.16.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,432 @@
...
+}
+
+pkg_postinst() {
+	local v
+	for v in ${REPLACING_VERSIONS}; do
+		if ver_test "${v}" -lt "1.9.0"; then
+			einfo "Note that vicmd_symbol config option was renamed to vimcmd_symbol in version 1.9"
+		fi
+	done
+}
```
Comment (juippis) [inline app-shells/starship/starship-1.16.0.ebuild:432]: We can probably get rid of this already?

**Review by juippis — APPROVED** [review]
I also don't see a reson to drop ~arm64 keywords, I'll ignore the keyword changes and merge. Thanks!

**Comment by xz-dev** [comment]
> I also don't see a reson to drop ~arm64 keywords, I'll ignore the keyword changes and merge. Thanks!

Oh, I got it. My bad, I forget `KEYWORDS="amd64 ~arm64"`, I will add it.

**Comment by juippis** [comment]
No need, I've already merged this. Will push in a minute with my own package updates.

And you want to reset keywords for amd64, so it becomes ~amd64. Generally when doing version bumps, you use **ekeyword** to reset keywords. **ekeyword ~all starship-1.16.0.ebuild**.

**Comment by xz-dev** [comment]
> No need, I've already merged this. Will push in a minute with my own package updates.
> 
> And you want to reset keywords for amd64, so it becomes ~amd64. Generally when doing version bumps, you use **ekeyword** to reset keywords. **ekeyword ~all starship-1.16.0.ebuild**.

Great, thank you for your explain.

File: app-shells/starship/starship-1.16.0.ebuild (hunk 3 above)
Comment (xz-dev) [reply app-shells/starship/starship-1.16.0.ebuild:432]: Yeah, I think so, it's an outdated message

**Pull request closed without merging**